include/nvexec/detail/memory.cuh

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../../stdexec/execution.hpp"
+
+#include <memory_resource>
+#include <new>
+#include <type_traits>
+
+#include "config.cuh"
+#include "throw_on_cuda_error.cuh"
+
+namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
+
+  struct device_deleter {
+    template <class T>
+    void operator()(T* ptr) {
+      ptr->~T();
```

Comment:
This is a new behavior that's shouldn't be necessary for this PR.

Reply:
I've commented it out. Is your concern that the destructor might get called from a place where the object is inaccessible?

Reply:
It's the same concern as in the following [comment](https://github.com/NVIDIA/stdexec/pull/952#discussion_r1222674960). Potential managed memory traffic.